### PR TITLE
Draft initial logic for user/schedule paging

### PR DESCRIPTION
**Pull request by matiasb**

### CODE REVIEW

File: engine/apps/api/serializers/paging.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    title = serializers.CharField(required=False, default=None)
+    message = serializers.CharField(required=False, default=None)
+
+    def validate(self, attrs):
+        if len(attrs["users"]) == 0 and len(attrs["schedules"]) == 0:
+            raise serializers.ValidationError("Provide at least one user or schedule")
+
+        if not attrs["alert_group_id"] and not attrs["title"]:
+            raise serializers.ValidationError("Provide either alert_group_id or title")
```
Comment (matiasb) [inline engine/apps/api/serializers/paging.py:54]: I think the idea was to fallback to a default title if none is provided? (the backend logic will set one if this is `None`).

**Review by matiasb — COMMENTED** [review]
FYI, will leave the integration update (ie. not using the existing manual one) and the escalation support for a next PR to avoid adding more complexity here (I have some WIP). Added one minor comment, API changes look good to me.

File: engine/apps/alerts/escalation_snapshot/escalation_snapshot_mixin.py [hunk 2]
```diff
@@ -253,6 +253,8 @@ def start_escalation_if_needed(self, countdown=ScenarioStep.CROSS_ACTION_DELAY,
 
         logger.debug(f"Start escalation for alert group with pk: {self.pk}")
 
+        # TODO: allow to start escalation at specific step?
```
Comment (vadimkerr) [inline engine/apps/alerts/escalation_snapshot/escalation_snapshot_mixin.py:256]: Do we plan to implement it as a part of this PR?

File: engine/apps/alerts/paging.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+def _trigger_alert(organization: Organization, team: Team, title: str, message: str, from_user: User) -> AlertGroup:
+    """Trigger manual integration alert from params."""
+    alert_receive_channel = AlertReceiveChannel.get_or_create_manual_integration(
+        organization=organization,
+        team=team,
+        integration=AlertReceiveChannel.INTEGRATION_MANUAL,
+        deleted_at=None,
+        defaults={
+            "author": from_user,
+            "verbal_name": f"Manual incidents ({team.name if team else 'General'} team)",
```
Comment (vadimkerr) [inline engine/apps/alerts/paging.py:29]: Probably change "incidents" to "alert groups"?

File: engine/apps/api/serializers/paging.py (hunk 1 above)
Comment (vadimkerr) [reply engine/apps/api/serializers/paging.py]: https://github.com/grafana/oncall/pull/1098/commits/55d248f9e612e1c47411f2ea2da27520d6acdee7

File: engine/apps/alerts/escalation_snapshot/escalation_snapshot_mixin.py (hunk 2 above)
Comment (matiasb) [reply engine/apps/alerts/escalation_snapshot/escalation_snapshot_mixin.py:256]: No, will leave this for later, will remove the comment for now.

File: engine/apps/alerts/paging.py (hunk 3 above)
Comment (matiasb) [reply engine/apps/alerts/paging.py:29]: Yeah, sounds good, changing.

**Pull request merged**